community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/NeoStoreFileListing.java

```diff
@@ -66,77 +76,56 @@ public NeoStoreFileListing( File storeDir, LabelScanStore labelScanStore,
+            {
+                files.add( toNotAStoreTypeFile.apply( file ) );
+            }
+            else if ( includeLogs && transactionLogFile( file.getName() ) )
+            {
+                files.add( toNotAStoreTypeFile.apply( file ) );
+            }
+        }
+    }
+
+    private Resource gatherLegacyIndexFiles( Collection<StoreFileMetadata> files ) throws IOException
     {
         final Collection<ResourceIterator<File>> snapshots = new ArrayList<>();
         for ( IndexImplementation indexProvider : legacyIndexProviders.all() )
         {
             ResourceIterator<File> snapshot = indexProvider.listStoreFiles();
             snapshots.add( snapshot );
-            Iterators.addToCollection( snapshot, files );
+            snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> files ) );
         }
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return new MultiResource( snapshots );
     }
 
-    private Resource gatherSchemaIndexFiles(Collection<File> targetFiles) throws IOException
+    private Resource gatherSchemaIndexFiles( Collection<StoreFileMetadata> targetFiles ) throws IOException
     {
         ResourceIterator<File> snapshot = indexingService.snapshotStoreFiles();
-        Iterators.addToCollection(snapshot, targetFiles);
+        snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> targetFiles ) );
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return snapshot;
     }
 
-    private Resource gatherLabelScanStoreFiles( Collection<File> targetFiles ) throws IOException
+    private Resource gatherLabelScanStoreFiles( Collection<StoreFileMetadata> targetFiles ) throws IOException
     {
         ResourceIterator<File> snapshot = labelScanStore.snapshotStoreFiles();
-        Iterators.addToCollection(snapshot, targetFiles);
+        snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> targetFiles ) );
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return snapshot;
     }
 
-    private void gatherNeoStoreFiles( final Collection<File> targetFiles, boolean includeTransactionLogs )
+    private void gatherNeoStoreFiles( final Collection<StoreFileMetadata> targetFiles )
     {
-        File neostoreFile = null;
-        for ( File dbFile : Objects.requireNonNull( storeDir.listFiles() ) )
-        {
-            String name = dbFile.getName();
-            if ( dbFile.isFile() )
-            {
-                if ( name.equals( MetaDataStore.DEFAULT_NAME ) )
-                {   // Keep it, to add last
-                    neostoreFile = dbFile;
-                }
-                else if ( neoStoreFile( name ) )
```

Comment:
By the looks of it this (now deleted) method included both counts store files, whereas the new implementation above only includes the current one. Is that correct behaviour and do we know what behavioural differences that would introduce?